File: 2019/day9/day9.py
#!/usr/bin/env python3
# https://adventofcode.com/2019/day/9

from pathlib import Path
import sys

sys.path.append(Path(__file__).parent.parent.as_posix())
from intcode.Intcode import Computer


filename = ("test.txt" if sys.argv[1] == "-t" else sys.argv[1]) if len(sys.argv) > 1 else "input.txt"
data = Path(filename).read_text()

computer = Computer()
computer.load(data)

computer.input.append(1)
computer.run()
print(computer.output.popleft())

computer.input.append(2)
computer.run()
print(computer.output.popleft())

File: 2019/day9/day9.rs
//! [Day 9: Sensor Boost](https://adventofcode.com/2019/day/9)

use intcode::{Computer, State};

fn main() {
    let args = aoc::parse_args();

    let mut program = Computer::load(&args.input);

    let mut run = |init| {
        program.reset();
        program.push(init);
        let mut result = 0;
        while let State::Output(value) = program.run() {
            result = value;
        }
        result
    };

    println!("{}", run(1)); // part 1
    println!("{}", run(2)); // part 2
}
